web-console/knikubevirt/Affinity/affinity.md

```diff
@@ -0,0 +1,60 @@
+# Affinity
+
+### About Affinity
+Affinity provieds the user to create a relationship between two resources chen scheduling them on the cluster.
+There are two high level relationship types:
+Affinity - attaching a group of resources together, making them run together in the same environment.
+Anti-affinity - creating a rejection relationship between a group of resources, thus avoiding sheduling any of them to the same evironment.
+
+Besides defining if we want to attach or reject resources to one another, we can choose to "Force" this rule we are creating.
+"Forcing" a rule means what the conditions of this rule must met, even if it means that some resources just won't be scheduled at all.
+
+### a few techincal words
+Affinity rules are Pods Property, written in the pod deployment file. When a pod is scheduled, the deployment file mentions that there is an affinity rule conditions which needs to be met.
+Selecting resources to affinity is usually done via labels. This makes sense since objects within k8 are temporary, and usually the way to target the same resource which keeps on crashing and regenerating is by it's label.
+
+
+## affinity rules - list page
+
+![Affinity list](img/D2-0-0.jpg)
+All affinity rules within the system can be viewed and managed from this page. It is located in the Master side navigation, in 'Workloads' under 'Secrets'
+
+## creating an affinity rule from Affinity Rules page
+
+Clicking on the "Add affinity rule" button at the top of the affinity list will open up the "Add affinty rule" modal.
+
+![Affinity rule modal - default](img/D2-1-0.jpg)
```

Comment:
This checkbox sounds scary. I wonder if we could simplify it or have a tooltip to better describe the implications of it.

Reply:
I rather stick to a checkbox, but a toggle might be easier to get:
WDYT?
![D2-1-0A](https://user-images.githubusercontent.com/48942365/60789789-992c5180-a168-11e9-8a17-9812f9006dd7.jpg)
![D2-1-0B](https://user-images.githubusercontent.com/48942365/60789790-99c4e800-a168-11e9-8936-a193dfa3e668.jpg)

Reply:
hmmm interesting. I like the simplicity of the checkbox too, but the radio does provide more context. 
My concern with the checkbox is left unchecked is it clear that they've chosen preferred?
What if the radio was "Rule conditions: Preferred or Required"?
I'm still wondering if adding an info tooltip could be helpful here too. Do we need to define what these things mean more clearly? I could see confusion on the users part if they aren't 100% familiar with affinity strictness.
 - Preferred "Will be apply this rule if possible but will not stop it from being scheduled"
 - Required "This resource will not be scheduled if rule is not met"
Does it feel like too much information to throw at the user?

Reply:
Looking at the two options (updated here according to your feedback) I can't determine which one is better:
![D2-1-0A](https://user-images.githubusercontent.com/48942365/60869807-55a11880-a238-11e9-9558-327066180ea7.jpg)
![D2-1-0B](https://user-images.githubusercontent.com/48942365/60869808-55a11880-a238-11e9-9c5d-f46a1d9f1ad2.jpg)

This is entering the realm of proper A/B user testing :)
I did run these with some random (non-tech savvy) people and apparently, the checkbox was clearer to them. The radio button option made them start to actually think about 'What is the difference between the two options'. So they lingered on this field trying to figure out something that is technically beyond them. The checkbox was more of a no-brainer since they felt it's an "Advanced" settings that probably needed to be left to default.

The main issue I have with the checkbox is on the List view, were becoming difficult to describe the user's selection. Like, what is the title, and what do I mark when it is not checked?
Other than that, I'm not worried that the user will get that this rule is "Preferred" since we are not talking in that terminology (We already somewhat drifted from the K8 terminology anyway)
If the list view makes sense then I think checkbox is better.

Of course, we can continue searching for the right copy for the radio buttons.

Reply:
I'm leaning towards the checkbox with the help text. (we can probably align the design a bit more with PF4 helper text).
Curious what others opinions are on this. @lizsurette @yfrimanm @yuvalgalanti 
I see this as an important feature to understand with affinity so it will be great to think it through. Could be a good case for some user testing too.